### PR TITLE
Add test cases for Add and Edit

### LINKED ISSUE #58: Test cases for Add and Edit commands

### PR DESCRIPTION
**Pull request by marraknas**
Fixes #58

### CODE REVIEW

File: src/test/java/seedu/address/model/person/MajorTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Test
+    public void constructor_invalidMajor_throwsIllegalArgumentException() {
+        String invalidMajor = "invalid";
+        assertThrows(IllegalArgumentException.class, () -> new Major(invalidMajor));
+    }
+
+    @Test
+    public void isValidMajor() {
```
Comment (PeterHW963) [inline src/test/java/seedu/address/model/person/MajorTest.java:24]: Based on naming convention for test cases, if you are testing whether single function is able to handle the different conditionals, it should have a test in front of it for future clarity and readibility. 
e.g. (from the website Week 3 topics)
@Test
    public void testStringConversion() {
        assertEquals("4,7", new IntPair(4, 7).toString());
    } 
    
 when the function name is stringConversion(). Likewise, here it should be named testIsValidMajor()

File: src/test/java/seedu/address/model/person/MajorTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        assertFalse(Major.isValidMajor("ce")); // incomplete major
+
+        // valid majors
+        assertTrue(Major.isValidMajor("cs"));
+        assertTrue(Major.isValidMajor("bza"));
+        assertTrue(Major.isValidMajor("ceg"));
+    }
+
+    @Test
+    public void equals() {
```
Comment (PeterHW963) [inline src/test/java/seedu/address/model/person/MajorTest.java:40]: same feedback here, should be testEquals()

File: src/test/java/seedu/address/model/person/RoleTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Test
+    public void constructor_invalidRole_throwsIllegalArgumentException() {
+        String invalidRole = "invalid";
+        assertThrows(IllegalArgumentException.class, () -> new Role(invalidRole));
+    }
+
+    @Test
+    public void isValidRole() {
```
Comment (PeterHW963) [inline src/test/java/seedu/address/model/person/RoleTest.java:24]: testIsValidRole()

File: src/test/java/seedu/address/model/person/RoleTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        assertFalse(Role.isValidRole("12345")); // numbers
+        assertFalse(Role.isValidRole("inval")); // incomplete role
+
+        // valid roles
+        assertTrue(Role.isValidRole("brUdder"));
+        assertTrue(Role.isValidRole("mUdder"));
+    }
+
+    @Test
+    public void equals() {
```
Comment (PeterHW963) [inline src/test/java/seedu/address/model/person/RoleTest.java:39]: testEquals()

**Review by PeterHW963 — COMMENTED** [review]
Logic wise LGTM!! Just naming conventions from what we were taught in the notes. I guess it's benefit is so that future us don't get confused seeing too many occurrences of equals() function when we search the entire directory.

File: src/test/java/seedu/address/model/person/MajorTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        assertEquals(major1.hashCode(), major2.hashCode());
+    }
+
+    @Test
+    public void hashCode_differentMajorDifferentHashCode() {
+        Major major1 = new Major("cs");
+        Major major2 = new Major("bza");
+
+        // different values -> different hashcode
+        assertFalse(major1.hashCode() == major2.hashCode());
```
Comment (psh12320) [inline src/test/java/seedu/address/model/person/MajorTest.java:74]: Following from the `hashCode_sameMajorSameHashCode()` method, I think you should consider using `assertNotEquals()` instead of `assertFalse()` here.

File: src/test/java/seedu/address/model/person/RoleTest.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        assertEquals(role1.hashCode(), role2.hashCode());
+    }
+
+    @Test
+    public void hashCode_differentRoleDifferentHashCode() {
+        Role role1 = new Role("mUdder");
+        Role role2 = new Role("brUdder");
+
+        // different values -> different hashcode
+        assertFalse(role1.hashCode() == role2.hashCode());
```
Comment (psh12320) [inline src/test/java/seedu/address/model/person/RoleTest.java:73]: Same as above, `assertNotEquals()` can be used here instead

File: src/test/java/seedu/address/model/person/MajorTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        String invalidMajor = "invalid";
+        assertThrows(IllegalArgumentException.class, () -> new Major(invalidMajor));
+    }
+
+    @Test
+    public void isValidMajor() {
+        // null major
+        assertThrows(NullPointerException.class, () -> Major.isValidMajor(null));
+
+        // invalid majors
```
Comment (psh12320) [inline src/test/java/seedu/address/model/person/MajorTest.java:28]: I think you should include test for a single or multiple whitespace, since it would likely be a very common occurence when using this app.

File: src/test/java/seedu/address/model/person/RoleTest.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        assertThrows(IllegalArgumentException.class, () -> new Role(invalidRole));
+    }
+
+    @Test
+    public void isValidRole() {
+        // null role
+        assertThrows(NullPointerException.class, () -> Role.isValidRole(null));
+
+        // invalid roles
+        assertFalse(Role.isValidRole("")); // empty string
```
Comment (psh12320) [inline src/test/java/seedu/address/model/person/RoleTest.java:29]: Same suggestion to include tests for whitespace as stated above.

**Review by psh12320 — COMMENTED** [review]
Overall LGTM!

**Review by XinghuaJulia — COMMENTED** [review]
LGTM! Minor nits as pointed out by Peter. Great job!!

**Review by ulricolo7 — COMMENTED** [review]
Overall LGB- I mean, LGTM!
